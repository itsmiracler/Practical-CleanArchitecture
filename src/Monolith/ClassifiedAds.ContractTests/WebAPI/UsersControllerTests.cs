using ApprovalTests;
using ApprovalTests.Reporters;
using ClassifiedAds.WebAPI.Models.Users;
using NJsonSchema;
using Xunit;

namespace ClassifiedAds.ContractTests.WebAPI
{
    [UseReporter(typeof(VisualStudioReporter))]
    public class UsersControllerTests
    {
        [Fact]
        public void UserModelTest()
        {
            var schema = JsonSchema.FromType<UserModel>().ToJson();
            Approvals.Verify(schema);
        }
    }
}
