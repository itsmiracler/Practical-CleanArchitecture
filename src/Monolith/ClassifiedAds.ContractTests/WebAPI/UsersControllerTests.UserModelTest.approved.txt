{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "UserModel",
  "type": "object",
  "additionalProperties": false,
  "properties": {
    "Id": {
      "type": "string",
      "format": "guid"
    },
    "UserName": {
      "type": [
        "null",
        "string"
      ]
    },
    "NormalizedUserName": {
      "type": [
        "null",
        "string"
      ]
    },
    "Email": {
      "type": [
        "null",
        "string"
      ]
    },
    "NormalizedEmail": {
      "type": [
        "null",
        "string"
      ]
    },
    "EmailConfirmed": {
      "type": "boolean"
    },
    "PhoneNumber": {
      "type": [
        "null",
        "string"
      ]
    },
    "PhoneNumberConfirmed": {
      "type": "boolean"
    },
    "TwoFactorEnabled": {
      "type": "boolean"
    },
    "ConcurrencyStamp": {
      "type": [
        "null",
        "string"
      ]
    },
    "SecurityStamp": {
      "type": [
        "null",
        "string"
      ]
    },
    "LockoutEnabled": {
      "type": "boolean"
    },
    "LockoutEnd": {
      "type": [
        "null",
        "string"
      ],
      "format": "date-time"
    },
    "AccessFailedCount": {
      "type": "integer",
      "format": "int32"
    },
    "Password": {
      "type": [
        "null",
        "string"
      ]
    }
  }
}